Src/WindowsServer/WindowsServer.Shared/Implementation/WindowsServerEventSource.cs

```diff
@@ -326,6 +326,71 @@ public void AzureInstanceMetadataFailureSettingDefaultPayload(string exception,
                 this.ApplicationName);
         }
 
+        [Event(33,
+            Message = "App Services Heartbeat Provider: Failed to obtain Azure App Services environment variable '{0}'. Exception raised: {1}",
+            Level = EventLevel.Informational)]
+        public void AppServiceHeartbeatPropertyAquisitionFailed(string envVarName, string exceptionStr, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(
+                33,
+                envVarName ?? "unknown",
+                exceptionStr ?? "unknown-exception",
+                this.ApplicationName);
+        }
+
+        [Event(34,
+            Message = "App Services Heartbeat Provider: Could not obtain the Heartbeat Manager instance during initialization. Exception raised: {0}",
+            Level = EventLevel.Warning)]
+        public void AppServiceHeartbeatManagerAccessFailure(string exceptionStr, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(
+                34,
+                exceptionStr ?? "unknown-exception",
+                this.ApplicationName);
+        }
+        
+        [Event(35,
+            Message = "App Services Heartbeat Provider: Accessing the Hearbeat Manager failed as it is not in the list of available modules.",
+            Level = EventLevel.Informational)]
```

Comment:
This could be Warning just like the above is warning.

Reply:
All these messages are Warning I think.